test/sets.jl

```diff
@@ -709,3 +709,20 @@ Base.IteratorSize(::Type{<:OpenInterval}) = Base.SizeUnknown()
     @test 3 ∈ i
     @test issubset(3, i)
 end
+
+@testset "IdSet" begin
+    A = Base.IdSet{Int}(1:4)
+    @test !isempty(A)
+    B = copy(A)
+    @test A ⊆ B
+    @test B ⊆ A
+    A = filter!(isodd, A)
+    @test A ⊆ B
+    @test !(B ⊆ A)
+    @test !isempty(A)
+    a = pop!(A, 1)
+    @test a == 1
+    @test !isempty(A)
+    A = empty!(A)
+    @test isempty(A)
+end
```

Comment:
Would be good to test that elements are compared with `===`, e.g. using an `IdSet{Vector{Int}}`.

Reply:
Better now?

Reply:
Maybe something like
```julia
julia> s = Base.IdSet{Vector{Int}}([a, b])
Base.IdSet{Array{Int64,1}} with 2 elements:
  [1]
  [2]

julia> a ∈ s
true

julia> c = [1]
1-element Array{Int64,1}:
 1

julia> c ∈ s
false
```

Reply:
Would probably be good to also add something that is `== a`, but not `=== a` to test that it's not just a regular `Dict`.

Reply:
To make such a test, I guess one could e.g. use `Vector{Int}` keys:
```julia
julia> dict = IdDict([0] => 1, [0] => 2)
IdDict{Array{Int64,1},Int64} with 2 entries:
  [0] => 1
  [0] => 2
```

Reply:
That said, this PR looks like a nice improvement, so even without that, it'd be nice to merge it (and improve it later?) Depending on whether @kshyatt still has any interest on "finishing" this or not, it'd be a shame to let this go to waste?